core/src/components/select/select.tsx

```diff
@@ -696,17 +730,170 @@ export class Select implements ComponentInterface {
 
   private renderLabel() {
     const { label } = this;
-    if (label === undefined) {
-      return;
-    }
 
     return (
-      <div class="label-text-wrapper">
-        <div class="label-text">{this.label}</div>
+      <div
+        class={{
+          'label-text-wrapper': true,
+          'label-text-wrapper-hidden': !this.hasLabel,
+        }}
+      >
+        {label === undefined ? <slot name="label"></slot> : <div class="label-text">{label}</div>}
       </div>
     );
   }
 
+  componentDidRender() {
+    if (this.needsExplicitNotchWidth()) {
+      /**
+       * Run this the frame after
+       * the browser has re-painted the select.
+       * Otherwise, the label element may have a width
+       * of 0 and the IntersectionObserver will be used.
+       */
+      raf(() => {
+        this.setNotchWidth();
+      });
+    }
+  }
+
+  /**
+   * Gets any content passed into the `label` slot,
+   * not the <slot> definition.
+   */
+  private get labelSlot() {
+    return this.el.querySelector('[slot="label"]');
+  }
+
+  /**
+   * Returns `true` if label content is provided
+   * either by a prop or a content. If you want
+   * to get the plaintext value of the label use
+   * the `labelText` getter instead.
+   */
+  private get hasLabel() {
+    return this.label !== undefined || this.labelSlot !== null;
+  }
+
+  private needsExplicitNotchWidth() {
+    if (
+      /**
+       * If the notch is not being used
+       * then we do not need to set the notch width.
+       */
+      this.notchSpacerEl === undefined ||
+      /**
+       * If no label is being used, then we
+       * do not need to estimate the notch width.
+       */
+      !this.hasLabel ||
+      /**
+       * If the label property is being used
+       * then we can render the label text inside
+       * of the notch and let the browser
+       * determine the notch size for us.
+       */
+      this.label !== undefined
```

Comment:
Optional: As far as the `label` checks go, `!this.hasLabel || this.label !== undefined` is equivalent to `this.label === undefined || this.label !== undefined`, which is always true, so you could combine these checks into `this.labelSlot === null`. I can understand wanting to black-box the `hasLabel` logic in case it changes in the future, though, so feel free to leave this as-is.

Reply:
Ah, true! Sure, that works for me.

Reply:
updated in https://github.com/ionic-team/ionic-framework/pull/27545/commits/ccc3b601d5f1305131496d3e02f6e43e0f4938de